auto_nag/scripts/topcrash_add_keyword.py

```diff
@@ -0,0 +1,165 @@
+
+        has_startup_signature = any(
+            self.topcrashes.get(signature) for signature in signatures
+        )
+
+        should_add_keyword = (
+            has_topcrash_signature
+            and not (has_topcrash_keyword or has_startup_signature)
+        ) or (has_startup_signature and not has_startup_keyword)
+
+        if not should_add_keyword:
+            return
+
+        keyword_to_add = "topcrash-startup" if has_startup_signature else "topcrash"
+
+        autofix = {
+            "keywords": {
+                "add": keyword_to_add,
+            },
+            "comment": {
+                "body": "The bug is linked to a startup topcrash signature."
+                if has_startup_signature
+                else "The bug is linked to a topcrash signature.",
+            },
+        }
+
+        ni_person = utils.get_mail_to_ni(bug)
+        if ni_person and bug["severity"] not in HIGH_SEVERITY:
+            autofix["flags"] = [
+                {
+                    "name": "needinfo",
+                    "requestee": ni_person["mail"],
+                    "status": "?",
+                    "new": "true",
+                }
+            ]
+            autofix["comment"]["body"] += (
+                f'\n:{ ni_person["nickname"] },'
+                "could you consider increasing the severity of this top-crash bug?"
+            )
+
+        autofix["comment"]["body"] += f"\n\n{ self.get_documentation() }\n"
+        self.autofix_changes[bugid] = autofix
+
+        data[bugid] = {
+            "severity": bug["severity"],
+            "added_keyword": keyword_to_add,
+        }
+
+        return bug
+
+    def get_bugs(self, date="today", bug_ids=[], chunk_size=None):
+        self.query_url = None
+        timeout = self.get_config("bz_query_timeout")
+        bugs = self.get_data()
+        params_list = self.get_bz_params(date)
+
+        searches = [
+            Bugzilla(
+                params,
+                bughandler=self.bughandler,
+                bugdata=bugs,
+                timeout=timeout,
+            ).get_data()
+            for params in params_list
+        ]
+
+        for search in searches:
+            search.wait()
+
+        return bugs
+
+    def get_bz_params(self, date):
+        self.topcrashes = Topcrash(
+            minimum_crashes=50,
+            minimum_startup_crashes=20,
+        ).get_signatures(date)
+
+        fields = [
+            "triage_owner",
+            "assigned_to",
+            "severity",
+            "keywords",
+            "cf_crash_signature",
+        ]
+        params_base = {
+            "include_fields": fields,
+            "resolution": "---",
+            "keywords": ["topcrash", "topcrash-startup"],
+            "keywords_type": "nowords",
+            "n1": 1,
+            "f1": "longdesc",
+            "o1": "casesubstring",
+            "v1": "could you consider increasing the severity of this top-crash bug?",
```

Comment:
This is going to exclude bugs where we first add the keyword, then the keyword is removed as the crash becomes less frequent but then it spikes again.

Reply:
Since we already exclude topcrash and topcrash-startup bugs, perhaps we can avoid this additional filter?

Reply:
I recall dropping this part; I guess that I got it back by mistake. Now it is dropped in 03689583f0fc20a4dea1053a0e7112461f3aa5a8.